compiler/parser/RdrHsSyn.hs

```diff
@@ -164,6 +165,18 @@ import Module
 
 #include "HsVersions.h"
 
+isDamlGenerated :: HasOccName a => a -> Bool
+isDamlGenerated namedThing =
+  let nameStr = occNameString $ occName namedThing
+  in
+  or
+    [ "_templateLet_" `isPrefixOf` nameStr
```

Comment:
why do the other identifiers get a dollar sign but not this one?

Reply:
unlike other constructs, templateLet does not get erased, so it gets retrieved by a bunch of other sections, including things which try to parse source back out, which leads to errors

I think you're right that we'll want to remove templateLets sooner rather than later - I don't think anyone is doing anything with them